packages/Adapters/POD_C/src/DTK_POD_PointCloudEntitySet.cpp

```diff
@@ -0,0 +1,146 @@
+    , d_global_ids( global_ids )
+    , d_num_points( num_points )
+    , d_space_dim( space_dim )
+    , d_layout( layout )
+{ /* ... */ }
+
+//---------------------------------------------------------------------------//
+// Get the parallel communicator for the entity set.
+Teuchos::RCP<const Teuchos::Comm<int> >
+POD_PointCloudEntitySet::communicator() const
+{
+    return d_comm;
+}
+
+//---------------------------------------------------------------------------//
+// Return the largest physical dimension of the entities in the set. 
+int POD_PointCloudEntitySet::physicalDimension() const
+{
+    return d_space_dim
+}
+
+//---------------------------------------------------------------------------//
+// Given an EntityId, get the entity.
+void POD_PointCloudEntitySet::getEntity( const EntityId entity_id,
+                                         const int topological_dimension,
+                                         Entity& entity ) const
+{
+    DTK_REQUIRE( 0 == topological_dimension );
+    DTK_REQUIRE( 1 == std::count(&d_global_ids[0],
+                                 &d_global_ids[0] + d_num_points,
+                                 entity_id) );
+    
+    int local_id = std::distance( &d_global_ids[0],
+                                  std::find(&d_global_ids[0],
+                                            &d_global_ids[0] + d_num_points,
+                                            entity_id) );
+    
+    entity = POD_PointCloudEntity( d_cloud_coords,
+                                   d_num_points,
+                                   d_space_dim,
+                                   d_layout,
+                                   entity_id,
+                                   local_id,
+                                   d_comm->getRank() );
+}
+
+//---------------------------------------------------------------------------//
+// Get an iterator over a subset of the entity set that satisfies the given
+// predicate. 
+EntityIterator POD_PointCloudEntitySet::entityIterator(
+    const int topological_dimension,
+    const PredicateFunction& predicate ) const
+{
+    DTK_REQUIRE( 0 == topological_dimension );
+    
+    return POD_PointCloudEntityIterator( d_cloud_coords,
+                                         d_global_ids,
+                                         d_num_points,
+                                         d_space_dim,
+                                         d_layout,
+                                         d_comm->getRank(),
+                                         predicate );
+}
+
+//---------------------------------------------------------------------------//
+// Given an entity, get the entities of the given type that are adjacent to
+// it. 
+void POD_PointCloudEntitySet::getAdjacentEntities(
+    const Entity& entity,
+    const int adjacent_dimension,
+    Teuchos::Array<Entity>& adjacent_entities ) const
+{
+    // There is no adjacency information in a point cloud.
+    DTK_INSIST( false );
```

Comment:
Either would technically be valid I suppose. I think throwing here is fine so someone does not use this with an algorithm that needs adjacency information.